erdos/utils.py

```diff
@@ -1,11 +1,44 @@
 import logging
+import time
 from functools import wraps
+from threading import Thread
 
 _freq_called = set([])
 
 
+def deadline(*expected_args):
+    """
+    Deadline decorator to be used for restraining computation latency
+    :param duration: computation's duration constrain in ms
+    :param deadline_missing_callback: function to call when the deadline is missed
+    :return:
+    """
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            global task_finished, target_ending_time
```

Comment:
I think we shouldn't use global here, but safer solutions like queues. See the [https://stackoverflow.com/questions/19790570/using-a-global-variable-with-a-thread](https://stackoverflow.com/questions/19790570/using-a-global-variable-with-a-thread).